MINOR: FetchRequest.Builder maxBytes for version <3

edoardocomar opened the pull request:
The maxBytes field should be set to DEFAULT_RESPONSE_MAX_BYTES,
the same way as the constructor using the Struct does.

codeveloped with @mimaison

edoardocomar committed:
MINOR: FetchRequest.Builder maxBytes for version <3

The maxBytes field should be set to DEFAULT_RESPONSE_MAX_BYTES,
the same way as the constructor using the Struct does.

codeveloped with @mimaison

ijuma commented:
Great catch. Can you please file a JIRA for this since it has user impact and it would be good to include in the release notes. Also, can you please add a unit test and perhaps assert in a few places that the value is positive?

ijuma commented:
On second thought, it seems like this has no impact because we use the right value when we deserialize.

edoardocomar commented:
@ijuma correct it has no impact - unless you were writing code that did not expect the FetchRequest to be serialized/deserialized, which was my case, so I caught it

edoardocomar commented:
happy to merge?

ijuma commented:
Maybe just add a trivial unit test that verifies this?

edoardocomar committed:
MINOR: FetchRequest.Builder maxBytes for version <3

The maxBytes field should be set to DEFAULT_RESPONSE_MAX_BYTES,
the same way as the constructor using the Struct does.

Added unit test for regression check

edoardocomar commented:
@ijuma done thanks

edoardocomar committed:
MINOR: FetchRequest.Builder maxBytes for version <3

The maxBytes field should be set to DEFAULT_RESPONSE_MAX_BYTES,
the same way as the constructor using the Struct does.

Added unit test for regression check

edoardocomar commented:
@ijuma could this PR be merged?

mimaison commented:
@ijuma Sorry for hijacking this thread but I've got a question that is related.

Looking at ProduceRequest.Builder, it only allows to create requests with version >= 2. As far as I can tell, ProduceRequest v0, 1 and 2 all have the same body. So why putting this limitation ?

ijuma reviewed (approved):
LGTM.

ijuma commented:
@mimaison the builder classes were introduced for the client compatibility work. Since we only support brokers with versions 0.10.0.0 and higher, we added some checks if older protocol versions were used because that would imply a bug somewhere.

As you may be aware, the request/response protocol classes are internal to Kafka. That has 2 major implications:
1. There are no compatibility guarantees.
2. We focus on the functionality that makes sense for a particular Kafka release. If we never use a particular protocol version, we may simply not support it.

Does that make sense?

mimaison commented:
Yes that makes sense. Thank you
